### PR TITLE
#175: Adjusted logic of ICarolAdataAdapted to new ImportService

### PR DESCRIPTION
**Pull request by oskarhinc**

### CODE REVIEW

File: src/main/java/org/benetech/servicenet/adapter/icarol/RelationManager.java [hunk 1]
```diff
@@ -73,30 +68,58 @@ private static void logProgress(int itemNr, int size) {
...
 
-    private void saveServiceRelatedData(Location location, ICarolProgram program,
-                                        Service savedService) {
-        persistence.importEligibility(program, savedService);
+    private Optional<Location> getLocationToPersist(ImportData importData, ICarolSite site) {
+        return mapper.extractLocation(
+            site.getContactDetails(), site.getId(), importData.getProviderName())
+            .flatMap(extractedLocation -> {
+
+                extractedLocation.setPhysicalAddress(
```
Comment (wojtekbu) [inline src/main/java/org/benetech/servicenet/adapter/icarol/RelationManager.java]: empty line

File: src/main/java/org/benetech/servicenet/adapter/icarol/RelationManager.java [hunk 2]
```diff
@@ -73,30 +68,58 @@ private static void logProgress(int itemNr, int size) {
...
+                    mapper.extractPhysicalAddress(site.getContactDetails()).orElse(null));
+
+                extractedLocation.setPostalAddress(
```
Comment (wojtekbu) [inline src/main/java/org/benetech/servicenet/adapter/icarol/RelationManager.java]: empty line

File: src/main/java/org/benetech/servicenet/adapter/icarol/RelationManager.java [hunk 3]
```diff
@@ -73,30 +68,58 @@ private static void logProgress(int itemNr, int size) {
...
     }
 
-    private void saveEntitiesWithoutOrganization(ICarolDataToPersist data, ImportData importData) {
-        saveLocationsAndRelatedData(data.getSites(), data, importData, null, null);
-        saveServicesAndRelatedData(null, null, data.getPrograms(), importData);
+    private Optional<Service> getServiceToPersist(ImportData importData, ICarolProgram program) {
+        Service extractedService = mapper
+            .extractService(program, importData.getProviderName());
+
+        extractedService.setEligibility(mapper.extractEligibility(program).orElse(null));
```
Comment (wojtekbu) [inline src/main/java/org/benetech/servicenet/adapter/icarol/RelationManager.java]: https://github.com/benetech/ServiceNet/wiki/iCarol-Data-Mapping-(Eden-and-UWBA)#restrictions

Eligibility cannot be null

Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/adapter/icarol/RelationManager.java]: Indeed, that's why we set the whole Eligibility entity/relation to null. If mapper can't extract it, we shouldn't persist Empty eligibility.

File: src/main/java/org/benetech/servicenet/adapter/icarol/RelationManager.java (hunk 1 above)
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/adapter/icarol/RelationManager.java]: ok

**Pull request merged**